### PR TITLE
quilt3 metadata docs update

### PR DESCRIPTION
**Pull request by robnewman**
# Description
Add warnings in docstrings

# TODO
- [X] [Changelog](CHANGELOG.md) entry

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 ## Catalog, Lambdas
 !-->
 ## Python API
+* [Added] Add docstring warnings for metadata ([#3156](https://github.com/quiltdata/quilt/pull/3156))
```
Comment (sir-sigurd) [inline docs/CHANGELOG.md:11]: I don't think it worth to add changelog entry for this. Also "added" is for more like for new features.

File: api/python/quilt3/packages.py [hunk 2]
```diff
@@ -187,6 +187,12 @@ def meta(self):
     def set_meta(self, meta):
         """
         Sets the user_meta for this PackageEntry.
+
+        Args:
+            meta(dict): user level metadata dict
+
+        Warnings:
+            meta(dict) arg value will override any previously defined metadata dict
```
Comment (sir-sigurd) [inline api/python/quilt3/packages.py:195]: I think that's quite obvious for this case.

**Review by sir-sigurd — COMMENTED** [review]
I think what we want is mostly to highlight that calling `set_dir('.', ...)` overrides meta set by `set_meta()`.

File: docs/CHANGELOG.md (hunk 1 above)
Comment (robnewman) [reply docs/CHANGELOG.md:11]: Updated to [Changed]

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,7 @@
 ## Catalog, Lambdas
 !-->
 ## Python API
+* [Changed] Add docstring warnings for metadata ([#3156](https://github.com/quiltdata/quilt/pull/3156))
```
Comment (sir-sigurd) [inline docs/CHANGELOG.md:11]: We don't add changelog entries for doc changes.

```suggestion
```

File: docs/api-reference/Bucket.md [hunk 4]
```diff
@@ -127,8 +127,8 @@ __Arguments__
 
 __Returns__
 
-``list``: Return value structure has not yet been permanently decided
-Currently, it's a `tuple` of `list` objects, containing the
+```list```: Return value structure has not yet been permanently decided
```
Comment (sir-sigurd) [inline docs/api-reference/Bucket.md:130]: I don't see a reason why this file was modified.

Comment (robnewman) [reply docs/api-reference/Bucket.md:130]: My understanding is that it gets modified when running `python build.py` in the `gendocs` directory.

Comment (robnewman) [reply docs/api-reference/Bucket.md:130]: My old (original?) commit wasn't using the correct version of `pydoc-markdown` and so messed this up. I had to create a virtualenv to get the right version. Shouldn't be modified in the latest commit(s).

File: docs/api-reference/api.md [hunk 5]
```diff
@@ -7,23 +7,18 @@ Set or read the QUILT configuration.
...
+    quilt3.config(navigator_url='https://example.com')
 
 To set config values, call with one or more key=value pairs:
 
-```python
-    >>> quilt3.config(navigator_url='http://example.com',
-    ...               elastic_search_url='http://example.com/queries')
-```
+    import quilt3
+    quilt3.config(navigator_url='http://example.com', elastic_search_url='http://example.com/queries')
```
Comment (akarve) [inline docs/api-reference/api.md:21]: The kwargs are unnecessary and kinda confusing here and above.

File: docs/CHANGELOG.md [hunk 6]
```diff
@@ -28,7 +28,7 @@ Entries inside each section should be ordered by type:
 
 # 5.1.0 - 2022-12-09
 ## Python API
-* [Changed] Add docstring warnings for metadata ([#3156](https://github.com/quiltdata/quilt/pull/3156))
+* [Added] `dedupe` parameter for `Package.push()` ([#3181](https://github.com/quiltdata/quilt/pull/3181))
```
Comment (akarve) [inline docs/CHANGELOG.md:31]: This is a changelog backfill? Belongs in its own PR.

File: docs/api-reference/Bucket.md [hunk 7]
```diff
@@ -127,8 +127,8 @@ __Arguments__
 
 __Returns__
 
-```list```: Return value structure has not yet been permanently decided
-Currently, it's a ``tuple`` of ``list`` objects, containing the
+``list``: Return value structure has not yet been permanently decided
+Currently, it's a `tuple` of `list` objects, containing the
```
Comment (akarve) [inline docs/api-reference/Bucket.md:131]: No need to scare people. If the return type changes, the API semver should reflect that.
```suggestion
``list``: Returns a `tuple` of `list` objects containing the
```

File: docs/api-reference/Bucket.md [hunk 8]
```diff
@@ -127,8 +127,8 @@ __Arguments__
...
 `following`: (directory info, file/object info, delete markers).
```
Comment (akarve) [inline docs/api-reference/Bucket.md:132]: No need to scare people. If the return type changes, the API semver should reflect that.

```suggestion
``list``: Returns a `tuple` of `list` objects formatted as follows: (directory info, file/object info, delete markers)
```

File: api/python/quilt3/packages.py [hunk 9]
```diff
@@ -827,6 +830,9 @@ def set_dir(self, lkey, path=None, meta=None, update_policy="incoming"):
         Raises:
             PackageException: When `path` doesn't exist.
             ValueError: When `update_policy` is invalid.
+
+        Warnings:
+            meta(dict) will override any previously defined metadata
```
Comment (sir-sigurd) [inline api/python/quilt3/packages.py:835]: I think there is no sense in duplicate parameter type here.

File: docs/CHANGELOG.md (hunk 6 above)
Comment (robnewman) [reply docs/CHANGELOG.md:31]: This is a result of a recent merge with `master`

**Comment by robnewman** [comment]
I believe the `test-gendocs` is failing because it doesn't know how to render `Warnings` docstrings.

File: api/python/quilt3/packages.py [hunk 10]
```diff
@@ -827,6 +830,9 @@ def set_dir(self, lkey, path=None, meta=None, update_policy="incoming"):
         Raises:
             PackageException: When `path` doesn't exist.
             ValueError: When `update_policy` is invalid.
+
+        Warnings:
+            meta will override any previously defined metadata
```
Comment (akarve) [inline api/python/quilt3/packages.py:835]:
```suggestion
            Calling this function with a logical key of "." or "/" overwrites any existing package-level metadata
```

File: api/python/quilt3/packages.py [hunk 11]
```diff
@@ -1119,6 +1125,9 @@ def set(self, logical_key, entry=None, meta=None, serialization_location=None, s
 
         Returns:
             self
+
+        Warnings:
+            meta will override any previously defined metadata
```
Comment (akarve) [inline api/python/quilt3/packages.py:1130]: This is implied.
```suggestion
```

File: api/python/quilt3/packages.py [hunk 12]
```diff
@@ -187,6 +187,9 @@ def meta(self):
     def set_meta(self, meta):
         """
         Sets the user_meta for this PackageEntry.
+
+        Args:
+            meta(dict): user level metadata dict
```
Comment (akarve) [inline api/python/quilt3/packages.py:192]:
```suggestion
            meta(dict): object-level metadata
```

**Review by akarve — CHANGES_REQUESTED** [review]

File: docs/api-reference/Bucket.md (hunk 8 above)
Comment (robnewman) [reply docs/api-reference/Bucket.md]: Outdated comment - this is no longer in the PR

**Comment by drernie** [comment]
Doesn't seem worth reviving. If we care about this, just add a new PR:
> I think what we want is mostly to highlight that calling set_dir('.', ...) overrides meta set by set_meta().
